controllers/pipelines/provider_service_manager.go

```diff
@@ -77,18 +79,28 @@ func (sm ServiceManager) Get(ctx context.Context, owner *pipelinesv1.Provider) (
 
 func (sm ServiceManager) Construct(provider *pipelinesv1.Provider) *corev1.Service {
 	prefixedProviderName := fmt.Sprintf("provider-%s", provider.Name)
-
-	ports := []corev1.ServicePort{{
-		Name:       "http",
-		Port:       int32(sm.config.DefaultProviderValues.ServicePort),
-		TargetPort: intstr.FromInt(sm.config.DefaultProviderValues.ServicePort),
-		Protocol:   corev1.ProtocolTCP,
-	}}
+	matchLabels := map[string]string{AppLabel: prefixedProviderName}
+	labels := MapConcat(sm.config.DefaultProviderValues.Labels, matchLabels)
+	ports := []corev1.ServicePort{
+		{
+			Name:       "http",
+			Port:       int32(sm.config.DefaultProviderValues.ServicePort),
+			TargetPort: intstr.FromInt(sm.config.DefaultProviderValues.ServicePort),
+			Protocol:   corev1.ProtocolTCP,
+		},
```

Comment:
could create a private constructor function like so

```go
func tcpServicePort(name string, port int) corev1.ServicePort {
    return corev1.ServicePort{
        Name:       name,
        Port:       int32(port),
        TargetPort: intstr.FromInt(port),
        Protocol:   corev1.ProtocolTCP,
    }
}
```

and then in the list of ports just call

```go
ports := []corev1.ServicePort{
    tcpServicePort("http", sm.config.DefaultProviderValues.ServicePort),
    tcpServicePort("metrics", sm.config.DefaultProviderValues.MetricsPort),
}
```

which makes it a tad more readable/less duplication 🤷